### PR TITLE
Raise warning level from 3 to 4

### PR DESCRIPTION
**Pull request by Lmocinemod**
The changes to error codes in 9fd8076b6dbf8d1aec5811965f6d8edf3b4e6522 are probably worth reviewing.

### CODE REVIEW

File: thprac/src/thprac/thprac_gui_locale.cpp [hunk 1]
```diff
@@ -474,7 +474,7 @@ static ImWchar baseUnicodeRanges[] =
         if (__glocale_disabled) {
             if (__glocale_disabled == 7) {
                 MessageBoxW(nullptr, L"No font can be loaded.\nthprac will now terminate.", L"Fatal error", MB_OK | MB_ICONERROR);
-                TerminateProcess(GetCurrentProcess(), -1);
+                TerminateProcess(GetCurrentProcess(), ERROR_FILE_CORRUPT);
```
Comment (32th-System) [inline thprac/src/thprac/thprac_gui_locale.cpp:477]: I think it'd be better to fix any warnings caused by `TerminateProcess(GetCurrentProcess(), ...)` lines by either changing it to `ExitProcess` or (if possible), returning back to WinMain, which then returns, right there in the same commit. I wanna get rid of all the  `TerminateProcess(GetCurrentProcess(), ...)` lines anyways cause that's jank

File: thprac/src/thprac/thprac_launcher_cfg.cpp [hunk 2]
```diff
@@ -1044,34 +1047,43 @@ class THUpdate {
...
+            return ERROR_HTTP_INVALID_QUERY_REQUEST;
 
         std::vector<uint8_t> buffer;
         auto remSize = fileSize;
         progressCallback(0, fileSize);
+        // TODO: This code seems suspicious. Documentation for InternetReadFile
+        // says that it should be called "until [it] returns TRUE and [byteRet]
+        // equals zero." But this loop seems content to exit early by one call,
+        // as long as the EXPECTED number of bytes have all been read.
+        // This may or may not be an actual problem?
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_cfg.cpp:1071]: I literally just copied it from thcrap

File: thprac/src/thprac/thprac_launcher_cfg.cpp [hunk 3]
```diff
@@ -1044,34 +1047,43 @@ class THUpdate {
...
+                // TODO: Does this actually mean the connection was lost?
+                // Or could it just have been aborted?
+                return ERROR_INTERNET_DISCONNECTED;
             }
             buffer.resize(readSize);
             if (InternetReadFile(hFile, buffer.data(), readSize, &byteRet) == FALSE) {
-                return -6;
+                // TODO: Does this actually mean it was aborted?
+                // Or could it have simply been lost?
+                return ERROR_INTERNET_CONNECTION_ABORTED;
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_cfg.cpp:1086]: Look for similar code in thcrap_update and ask Bruno (brliron) instead

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 4]
```diff
@@ -140,19 +140,8 @@ class GuiWaitingAnm {
...
-    SHFILEOPSTRUCTW fileOp;
-    memset(&fileOp, 0, sizeof(SHFILEOPSTRUCTW));
-    fileOp.wFunc = FO_DELETE;
-    fileOp.pFrom = path.c_str();
-    fileOp.fFlags = FOF_NOCONFIRMATION | FOF_NOERRORUI | FOF_NOCONFIRMMKDIR | FOF_SILENT;
-    SHFileOperationW(&fileOp);
-}
-
-static void MovWndToTop(HWND m_hWnd)
+// TODO: Move to source file?
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:143]: Sure

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 5]
```diff
@@ -166,18 +155,8 @@ static void MovWndToTop(HWND m_hWnd)
...
-}
-static void GuiPopCursorPos()
-{
-    ImGui::SetCursorPos(g_guiCursorStack.back());
-    g_guiCursorStack.pop_back();
-}
-
-static void GuiColumnText(const char* text)
+// TODO: Move to source file?
+inline void GuiColumnText(const char* text)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:159]: yeah

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 6]
```diff
@@ -193,23 +172,27 @@ static void GuiColumnText(const char* text)
     }
 }
 
-static void GuiCenteredText(const char* text)
+// TODO: Move to source file?
+inline void GuiCenteredText(const char* text)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:176]: Not sure. This I would like to always be inlined

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 7]
```diff
@@ -193,23 +172,27 @@ static void GuiColumnText(const char* text)
...
 {
     ImGui::SetCursorPosX((ImGui::GetWindowWidth() - ImGui::CalcTextSize(text, 0, false, 0.0f).x) / 2.0f);
     ImGui::TextWrapped("%s", text);
 }
 
-static void GuiSetPosXText(const char* text, float offset = 0.0f)
+// TODO: Move to source file?
+inline void GuiSetPosXText(const char* text, float offset = 0.0f)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:183]: No

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 8]
```diff
@@ -193,23 +172,27 @@ static void GuiColumnText(const char* text)
...
 {
     ImGui::SetCursorPosX((ImGui::GetWindowWidth() - ImGui::CalcTextSize(text, 0, false, 0.0f).x) / 2.0f - offset);
 }
 
-static void GuiSetPosYRel(float rel)
+// TODO: Move to source file?
+inline void GuiSetPosYRel(float rel)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:189]: No

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 9]
```diff
@@ -193,23 +172,27 @@ static void GuiColumnText(const char* text)
...
 {
     ImGui::SetCursorPosY(ImGui::GetWindowHeight() * rel);
 }
 
-static void GuiHelpMarker(const char* desc)
+// TODO: Move to source file?
+inline void GuiHelpMarker(const char* desc)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:195]: perhaps

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 10]
```diff
@@ -221,36 +204,8 @@ static void GuiHelpMarker(const char* desc)
...
-            ImGui::TextUnformatted(descTxt);
-            ImGui::PopTextWrapPos();
-            ImGui::EndTooltip();
-        }
-    }
-}
-
-static int GuiCornerButton(const char* text, const char* text2 = nullptr, const ImVec2& offset = ImVec2(1.5f, 0.5f), bool useCurrentY = false)
+// TODO: Move to source file?
+inline int GuiCornerButton(const char* text, const char* text2 = nullptr, const ImVec2& offset = ImVec2(1.5f, 0.5f), bool useCurrentY = false)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:208]: def big enough to go in a source file

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 11]
```diff
@@ -281,15 +236,17 @@ static int GuiCornerButton(const char* text, const char* text2 = nullptr, const
     return result;
 }
 
-static bool GuiButtonRelCentered(const char* buttonText, float posYRel, const ImVec2& sizeRel)
+// TODO: Should this be moved to a source file?
+inline bool GuiButtonRelCentered(const char* buttonText, float posYRel, const ImVec2& sizeRel)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:240]: Not sure, I think I like this being inlined as well

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 12]
```diff
@@ -281,15 +236,17 @@ static int GuiCornerButton(const char* text, const char* text2 = nullptr, const
...
 {
     auto wndSize = ImGui::GetWindowSize();
     ImGui::SetCursorPosX((wndSize.x - wndSize.x * sizeRel.x) / 2.0f);
     ImGui::SetCursorPosY(wndSize.y * posYRel);
     return ImGui::Button(buttonText, ImVec2(wndSize.x * sizeRel.x, wndSize.y * sizeRel.y));
 }
 
-static bool GuiButtonTxtCentered(const char* buttonText, float posYRel)
+// TODO: Move to source file?
+inline bool GuiButtonTxtCentered(const char* buttonText, float posYRel)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:249]: and this

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 13]
```diff
@@ -318,35 +267,15 @@ static bool GuiModal(const char* modalTitle, ImVec2 sizeRel = ImVec2(0.0f, 0.0f)
     return ImGui::BeginPopupModal(modalTitle, nullptr, ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoMove);
 }
 
-static bool GuiModalFullScreen(const char* modalTitle)
+// TODO: Move to source file?
+inline bool GuiModalFullScreen(const char* modalTitle)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:271]: and this as well

File: thprac/src/thprac/thprac_launcher_utils.h [hunk 14]
```diff
@@ -375,7 +304,8 @@ static bool GuiButtonYesNo(const char* buttonText1 = "OK", const char* buttonTex
     return false;
 }
 
-static bool GuiButtonAndModalYesNo(const char* buttonText, const char* modalTitle, const char* modalText, float buttonSize = 6.0f, const char* buttonText1 = "OK", const char* buttonText2 = "Cancel")
+// TODO: Move to source file?
+inline bool GuiButtonAndModalYesNo(const char* buttonText, const char* modalTitle, const char* modalText, float buttonSize = 6.0f, const char* buttonText1 = "OK", const char* buttonText2 = "Cancel")
```
Comment (32th-System) [inline thprac/src/thprac/thprac_launcher_utils.h:308]: Maybe

File: thprac/src/thprac/thprac_main.cpp [hunk 15]
```diff
@@ -272,7 +271,7 @@ bool RunGameWithTHPrac(THGameSig& gameSig, std::wstring& name)
 
     auto result = (WriteTHPracSig(proc_info.hProcess) && THPrac::LoadSelf(proc_info.hProcess));
     if (!result)
-        TerminateThread(proc_info.hThread, -1);
+        TerminateThread(proc_info.hThread, ERROR_FUNCTION_FAILED);
```
Comment (32th-System) [inline thprac/src/thprac/thprac_main.cpp:274]: I think it'd be better to fix any warnings caused by `TerminateProcess(GetCurrentProcess(), ...)` lines by either changing it to `ExitProcess` or (if possible), returning back to WinMain, which then returns, right there in the same commit. I wanna get rid of all the  `TerminateProcess(GetCurrentProcess(), ...)` lines anyways cause that's jank

File: thprac/src/thprac/thprac_th14.cpp [hunk 16]
```diff
@@ -1014,25 +1013,26 @@ namespace TH14 {
...
-            std::wstring _msg = msg;
+            std::wstring finalMessage = msg;
             if (msg2) {
-                _msg += msg2;
+                finalMessage += msg2;
             }
-            MessageBoxW(nullptr, _msg.c_str(), _title, type);
+            MessageBoxW(nullptr, finalMessage.c_str(), _title, type);
         }
         void MsgBox(UINT type, const char* title, const char* msg, const char* msg2 = nullptr)
```
Comment (32th-System) [inline thprac/src/thprac/thprac_th14.cpp:1025]: TODO: use newer utf8_to_utf16 function

File: thprac/src/thprac/thprac_main.cpp (hunk 15 above)
Comment (Lmocinemod) [reply thprac/src/thprac/thprac_main.cpp:274]: I'm not sure if this comment was a duplicate or not, but the thread being terminated in this case seems to be the target game, not thprac itself.

Comment (32th-System) [reply thprac/src/thprac/thprac_main.cpp:274]: Oh, I must've misread and copy pasted the comment. Yeah in this case you'd call ExitThread

Comment (32th-System) [reply thprac/src/thprac/thprac_main.cpp:274]: Nevermind, I read that code completely wrong

**Pull request merged**